src/renderer/components/+extensions/extensions.tsx

```diff
@@ -38,8 +39,28 @@ interface InstallRequestValidated extends InstallRequestPreloaded {
   tempFile: string; // temp system path to packed extension for unpacking
 }
 
+type ExtensionState = "uninstalling";
+
+interface Props {}
+interface State {
+  extensionState: {
+    [lensExtensionId: string]: {
+      displayName: string;
+      state: ExtensionState
+    }
+  };
+}
+
+/**
+ * Extensions that were removed from extensions but are still in "uninstalling" state
+ */
+const getRemovedUninstalling = (extensions: InstalledExtension[], extensionState: State["extensionState"]) => 
+  Object.entries(extensionState).filter(([id, extensionState]) => 
+    extensionState.state === "uninstalling" && !extensions.find(extension => extension.id === id)
+  ).map(([id, extension]) => ({ ...extension, id }));
```

Comment:
top level items should be functions preferably (not bound anonymous functions).

Reply:
Changed